DateTimeCalendar/Date.java

```diff
@@ -0,0 +1,322 @@
+            y -= years;
+
+            if (m - months < 1){
+                while (m - months < 1){
+                    y--;
+                    months -= 12;
+                }
+            }
+            m -= months;
+
+            if (d - days < 1){
+                while(d - days < 1) {
+                    if (m - 1 < 1){
+                        y--;
+                        m = 12;
+                    } else {
+                        m--;
+                    }
+
+                    if (m == 2) {
+                        if (isLeapYear(y)) {
+                            days -= Date.Month.values()[m - 1].days + 1;
+                        }
+                    } 
+
+                    days -= Date.Month.values()[m - 1].days;
+                }
+            } 
+            d -= days;
+        } else {
+            throw new IllegalArgumentException("Invalid input");
+        }
+        
+        //Returning new object is a design decision. Unsure if new object or reference would be needed, to play it safe returning new obj. Example: returning vacation days might need to be a new hash.
+        return new Date(y, Date.Month.values()[m - 1], d);
+    }
+
+    public int diff(Date input){
+        //return in value of days
+        int payload = 0;
+        int currentDays = (int)convToDays(this.year, this.month.ordinal() + 1, this.day);
+        int inputDays = (int)convToDays(input.year, input.month.ordinal() + 1, input.day);
+
+        if (currentDays > inputDays){
+            payload = currentDays - inputDays;
+        } else {
+            payload = inputDays - currentDays;
+        }
+        return payload;
+    }
+
+    public Date add(int years, int months, int days){
+
+        if (years >= 0 && months >= 0 && days >= 0){
+            years += this.year;
+            months += this.month.ordinal() + 1;
+            days += this.day;
+
+            //handle Months
+            if (months > 12) {
+                years += months / 12;
+                months = months % 12;
+            }
+            //set months back to 0-11
+            months--;
+
+            while (days > 27){
+                int currentMonthDayCount = Date.Month.values()[months].days;
+                if (months == 1 && Date.isLeapYear(years)){
+                    currentMonthDayCount++;
+                }
+
+                if (days > currentMonthDayCount){
+                    if (days > DAYS_IN_REG_YEAR){
+                        if (isLeapYear(years)){
+                            days -= DAYS_IN_LEAP_YEAR;
+                        } else {
+                            days -= DAYS_IN_REG_YEAR;
+                        }
+                        years++;
+                    } else {
+                        days -= currentMonthDayCount;
+                        if (months == 11){
+                            months = 0;
+                        } else {
+                            months++;
+                        }
```

Comment:
enter the magic of the modulus operation. Remember that mod is a function that will always map `X` to a number between `[0, N]`, in this case you can do:
```java
months = (months + 1) % 12;
```

Reply:
Added this in (hehe get it?). I tried to do the same with the subtract method but ran into some challenges. I have a functional method in place but would like to figure out how to add the modules logic into the `Date.subtract` method.